### PR TITLE
ML Design Update & Validation Doc Updates & Ops Doc Updates

### PR DESCRIPTION
**Pull request by peterdeutsch**

### CODE REVIEW

File: docs/design/3_1_2_platform_management.tex [hunk 1]
```diff
@@ -11,14 +11,14 @@ \subsubsection{Subsystem Overview}
...
+The selected device to implement the platform management system is the Raspberry Pi 4. As seen in Figure \ref{rpi}, the Raspberry Pi 4 is a small single-board computer which hosts an ARM Cortex A72 CPU, a CSI camera interface, 40 serial pins, WiFi/Bluetooth, and a gigabit Ethernet port. The Raspberry Pi 4 was selected over other single-computer platforms (including Arduino/Beagleboard/Orange Pi) as it:
 \begin{itemize}
 \item is relatively inexpensive (CAN\$45 in November 2019),
 \item has a large support community and a wealth of driver/library support,
 \item supports a wide range of high throughput I/O standards (USB/WiFi/Ethernet/Serial, \textbf{C.EX.3}),
 \item provides an industry-standard camera interface (\textbf{C.EX.5}), and
 \item supports extensible non-volatile storage (\textbf{F.CP.4}).
-\item is regularly succeeded with cross-compatible revisions (such as a Raspberry Pi 5), future-proofing the PMB segment
+\item has been regularly succeeded with cross-compatible revisions (presumably, including the upcoming Raspberry Pi 5) -- future-proofing the PMB segment
```
Comment (FSXAC) [inline docs/design/3_1_2_platform_management.tex]: Maybe back these facts up with citations/datasheets/cross-reference validation documents?

File: docs/design/3_1_2_platform_management.tex [hunk 2]
```diff
@@ -35,18 +35,19 @@ \subsubsection{Data Flow}
...
+The presence of on-board WiFi/camera interfaces is the primary driving factor towards adopting a dual-board (Platform Management and Programmable) solution. While a single-board solution is possible, the development effort and cost of the computing platform would be considerably higher. Most programmable logic boards (including our chosen board, the Zedboard) require external expansion devices to interface with peripherals such as WiFi/cameras. These expansion devices, which connect to the programmable logic board via a serial connection, have poorly documented drivers and are of considerable expense -- ranging from US\$25 for a basic WiFi module\cite{digiwifi} to US\$200 for a compatible camera interface card\cite{digipmod}. External expansion devices also require considerable glue logic on the programmable logic board, potentially risking non-compliance with constraint \textbf{C.EX.2}.
```
Comment (FSXAC) [inline docs/design/3_1_2_platform_management.tex]:
```suggestion
The presence of on-board Wi-Fi and camera interfaces is the primary driving factor towards adopting a dual-board (PMB and PLB) solution. While a uni-board solution is possible, the development effort and cost of the computing platform would be considerably higher. Most programmable logic boards (including our chosen board, the Zedboard) require external expansion devices to interface with peripherals such as WiFi/cameras. These expansion devices, which connect to the programmable logic board via a serial connection, have poorly documented drivers and are of considerable expense -- ranging from US\$25 for a basic WiFi module\cite{digiwifi} to US\$200 for a compatible camera interface card\cite{digipmod}. External expansion devices also require considerable glue logic on the programmable logic board, potentially risking non-compliance with constraint \textbf{C.EX.2}.
```

Comment (FSXAC) [reply docs/design/3_1_2_platform_management.tex]: Additional reasoning:

- Everything fitted onto a single board would have longer turn-around time for design iterations
- Uni-board is harder to integrate with the drone as it is less modular --- reduces mechanical freedom
- Uni-board goes against the requirement/objective of expandability

**Comment by peterdeutsch** [comment]
Apologies for continuously tacking on unrelated commits to this PR, my local branch structure is screwy.

File: docs/design/3_1_3_programmable_logic.tex [hunk 3]
```diff
@@ -26,21 +26,41 @@ \subsubsection{Device Selection}
...
+The selection of this particular programmable logic board was mainly motivated by cost. The client was able to provide a ZedBoard unit free-of-charge, stemming from donations through the Xilinx University Program. While this board has a fairly small number of logical cells (85,000 LCs) compared to other programmable logic boards on the market, especially those which target video applications, the project's budget does not allow for the purchase of more performant boards without significant sacrifices to the multirotor assembly. The number of logical cells on the ZedBoard is sufficient to comply with the minimum LC constraint (\textbf{C.EX.1}). Additionally, the Xilinx Zynq-7000 SoC is ubiquitous in the enthusiast/academic market, providing a large amount of online resources to assist in development.
```
Comment (wzfftxwd) [inline docs/design/3_1_3_programmable_logic.tex]:
```suggestion
The selection of this particular programmable logic board was mainly motivated by cost. The client was able to provide a ZedBoard unit free-of-charge, stemming from donations through the Xilinx University Program. While this board has a fairly small number of logical cells (85,000 LCs) compared to other programmable logic boards on the market, especially those which target video applications, the project's budget does not allow for the purchase of more performant boards without significant sacrifices to the multirotor assembly. The number of logical cells on the ZedBoard is sufficient to comply with the minimum LC constraint (\textbf{C.EX.1}). Additionally, the Xilinx Zynq-7000 SoC is ubiquitous in the enthusiast/academic market, providing a large number of online resources to assist in development.
```

File: docs/design/3_1_3_programmable_logic.tex [hunk 4]
```diff
@@ -26,21 +26,41 @@ \subsubsection{Device Selection}
...
+At a high level, the YOLOv2 accelerator speeds up ML tasks through several optimization strategies. Most prominently, the YOLOv2 accelerator aggressively unrolls and pipelines implicit loops throughout the neural network\footnote{More information about loop unrolling can be found in Xilinx's \textit{Loop Pipelining and Loop Unrolling} guide\cite{xilinx}}. Other optimizations strategies employed include use of 16-bit fixed point representations, parameter re-arrangement, ping-pong buffering\cite{pingpong}, and communication through parallel data buses. The module itself is written in the C language, and is converted to a hardware representation during system configuration through the use of a high-level synthesis (HLS) tool.
```
Comment (wzfftxwd) [inline docs/design/3_1_3_programmable_logic.tex]:
```suggestion
At a high level, the YOLOv2 accelerator speeds up ML tasks through several optimization strategies. Most prominently, the YOLOv2 accelerator aggressively unrolls and pipelines implicit loops throughout the neural network\footnote{More information about loop unrolling can be found in Xilinx's \textit{Loop Pipelining and Loop Unrolling} guide\cite{xilinx}}. Other optimizations strategies employed include the use of 16-bit fixed point representations, parameter re-arrangement, ping-pong buffering\cite{pingpong}, and communication through parallel data buses. The module itself is written in the C language, and is converted to a hardware representation during system configuration through the use of a high-level synthesis (HLS) tool. 
```

File: docs/design/3_1_3_programmable_logic.tex [hunk 5]
```diff
@@ -26,21 +26,41 @@ \subsubsection{Device Selection}
...
+
+\textit{MARLANN} is an open-source IP core capable of performing 8-bit fixed point signed integer arithmetic, matrix multiplication, and max-pooling. While MARLANN has a low FPGA area overhead (approx. 5,000 LCs), its relative simplicity would necessitate a large amount of effort in developing glue logic to form a functional ML system. The MARLANN project is also not regularly maintained, leading to further concerns regarding future project maintenance/extensibility.
+
+\textit{FREE-TPU} is a free (MIT licensed) acceleration platform developed by EmbedEEP. While the platform allows for greater flexibility in model selection compared to our selected solution (allowing for more less complex models such as MobileNet-YOLO), the project is no longer maintained. In addition, as the source code is unavailable (the API is made available as a shared library), we are prohibited from making appropriate modifications to better suit our platform (ex. modifying image data structures), leading to unacceptably poor performance (less than 0.25 FPS).
```
Comment (wzfftxwd) [inline docs/design/3_1_3_programmable_logic.tex]: Is that a typo in second line? More less complex?

File: docs/design/3_1_3_programmable_logic.tex [hunk 6]
```diff
@@ -26,21 +26,41 @@ \subsubsection{Device Selection}
...
+
+\begin{figure}[H]
+\centering
+\includegraphics[width=14cm]{img/fpga_interconnect.pdf}
+\caption{FPGA Interconnect Block Diagram}
+\label{interconnect}
+\end{figure}
+
+\subsubsection{On-Chip Interconnects}
+As seen in Figure \ref{interconnect}, the YOLOv2 accelerator is interfaced through 5 parallel AXI buses -- four buses to transfer weight data and features (including video frames) \textit{to the accelerator} and one bus to transfer system signals/results \textit{to the HPS}. A C++ program running on the HPS coordinates these interactions. In addition, each AXI bus passes through an AXI SmartConnect block (Xilinx IP) to facilitate cross-domain clocking. This is required as the YOLOv2 accelerator is driven at an higher clock frequency (150MHz compared to the provided 50MHz global clock) to further accelerate ML computations. The higher clock frequency used by the accelerator is generated through the use of a phase-locked loop (PLL) -- an efficient and commonly used construct in circuit design. The PLL is shown in Figure \ref{interconnect} as a \textit{Clocking Wizard} block.
```
Comment (wzfftxwd) [inline docs/design/3_1_3_programmable_logic.tex]:
```suggestion
As seen in Figure \ref{interconnect}, the YOLOv2 accelerator is interfaced through 5 parallel AXI buses -- four buses to transfer weight data and features (including video frames) \textit{to the accelerator} and one bus to transfer system signals/results \textit{to the HPS}. A C++ program running on the HPS coordinates these interactions. In addition, each AXI bus passes through an AXI SmartConnect block (Xilinx IP) to facilitate cross-domain clocking. This is required as the YOLOv2 accelerator is driven at a higher clock frequency (150MHz compared to the provided 50MHz global clock) to further accelerate ML computations. The higher clock frequency used by the accelerator is generated through the use of a phase-locked loop (PLL) -- an efficient and commonly used construct in circuit design. The PLL is shown in Figure \ref{interconnect} as a \textit{Clocking Wizard} block.
```

File: docs/design/3_1_3_programmable_logic.tex (hunk 5 above)
Comment (peterdeutsch) [reply docs/design/3_1_3_programmable_logic.tex]:
```suggestion
\textit{FREE-TPU} is a free (MIT licensed) acceleration platform developed by EmbedEEP. While the platform allows for greater flexibility in model selection compared to our selected solution (allowing for less complex models such as MobileNet-YOLO), the project is no longer maintained. In addition, as the source code is unavailable (the API is made available as a shared library), we are prohibited from making appropriate modifications to better suit our platform (ex. modifying image data structures), leading to unacceptably poor performance (less than 0.25 FPS).
```

**Pull request merged**